Feature/create task

MahmoodTarek opened the pull request.

MahmoodTarek committed:
implement task creation logic in CreateTaskUseCase, TaskDataSourceImpl, and TaskRepositoryImpl

MahmoodTarek committed:
Merge remote-tracking branch 'origin/development' into feature/create-task

# Conflicts:
#	src/main/kotlin/data/repository/TaskRepositoryImpl.kt
#	src/main/kotlin/domain/usecase/task/CreateTaskUseCase.kt

MahmoodTarek committed:
Merge branch 'development' into feature/create-task

MahmoodTarek committed:
Merge remote-tracking branch 'origin/development' into feature/create-task

AbdElrahman-Rafaat-Amer commented on src/main/kotlin/domain/usecase/task/CreateTaskUseCase.kt:
```diff
@@ -1 +1,21 @@
+
+class CreateTaskUseCase(
+    val taskRepository: TaskRepository
+) {
```
U can just `return taskRepository.createTask(task)` as no new work needed here

AbdElrahman-Rafaat-Amer reviewed (approved):
Greeeat

AbdElrahman-Rafaat-Amer commented on src/main/kotlin/data/repository/TaskRepositoryImpl.kt:
```diff
@@ -11,7 +13,29 @@ class TaskRepositoryImpl(
+        return createdTask.fold(
+            onSuccess = {
+                val auditLog = AuditLog(
+                    auditId = UUID.randomUUID(),
```
`auditId = UUID.randomUUID(),`
it's already defined so no need to reassign it

MahmoodTarek committed:
resolve AppModule import issue
refactor CreateTaskUseCase.kt remove the fold
resolve the TaskDataSourceImpl.kt Issue
refactor TaskRepositoryImpl.kt remove reassign auditId

MahmoodTarek committed:
resolve AppModule import issue
refactor CreateTaskUseCase.kt remove the fold
resolve the TaskDataSourceImpl.kt Issue
refactor TaskRepositoryImpl.kt remove reassign auditId

MahmoodTarek committed:
resolve TaskDataSourceImplTest.kt issue the createTask with called by mistake
add to TaskMock.kt the validAuditLog
resolve TaskRepositoryImpl.kt to return the auditDataSource
resolve the TaskRepositoryImplTest.kt issue that with need an auditLog too

ALiyASSER0 reviewed (approved):
Good job

AbdElrahman-Rafaat-Amer committed:
Merge remote-tracking branch 'origin/development' into feature/create-task

# Conflicts:
#	src/main/kotlin/data/repository/TaskRepositoryImpl.kt
#	src/main/kotlin/data/storage/task/TaskDataSourceImpl.kt
#	src/main/kotlin/domain/usecase/task/CreateTaskUseCase.kt
#	src/test/kotlin/data/repository/TaskRepositoryImplTest.kt
#	src/test/kotlin/data/storage/task/TaskDataSourceImplTest.kt

AbdElrahman-Rafaat-Amer committed:
fix merge conflict when merge development into feature/create-task
